### PR TITLE
Rxjs task 9

### PR DESCRIPTION
**Pull request by gegorov**

### CODE REVIEW

File: src/app/core/interceptors/loader-interceptor/loader.interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from "@angular/common/http";
+import { Observable } from "rxjs";
+
+import { LoadingService } from "../../services/index";
+import { debounceTime, finalize } from "rxjs/operators";
+
+
+@Injectable()
+export class LoaderInterceptor implements HttpInterceptor {
+
```
Comment (kirillpanov) [inline src/app/core/interceptors/loader-interceptor/loader.interceptor.ts]: please, remove empty line

File: src/app/core/interceptors/loader-interceptor/loader.interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        if (this.activeRequests === 0) {
+            this.loadingService.startLoading();
+        }
+
+        this.activeRequests++;
+        return next.handle(request).pipe(
+            debounceTime(5000),
```
Comment (kirillpanov) [inline src/app/core/interceptors/loader-interceptor/loader.interceptor.ts]: I don't think it's a good idea to delay fetched data to user for 5 seconds

File: src/app/core/services/loading-service/loading-service.service.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { Injectable } from "@angular/core";
+import { BehaviorSubject } from "rxjs";
+
+@Injectable()
+export class LoadingService {
+    // tslint:disable-next-line:variable-name
+    private _loading: boolean = false;
+
+    get loading(): boolean {
```
Comment (kirillpanov) [inline src/app/core/services/loading-service/loading-service.service.ts]: public

File: src/app/core/services/loading-service/loading-service.service.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        return this._loading;
+    }
+
+    set loading(value: boolean) {
```
Comment (kirillpanov) [inline src/app/core/services/loading-service/loading-service.service.ts]: public

File: src/app/courses/courses-list/courses-list.component.ts [hunk 5]
```diff
@@ -61,12 +67,12 @@ export class CoursesListComponent implements OnInit {
...
 
     /**
-     * Function that receives search string from search component and filters courses
-     * @param value search string that is emitted by search component
+     * Function that receives search string from search component and makes api call for backend search
      */
-    public onSearchNotify(value: string): void {
-        this.courseService.nextQuery(value);
-    }
+    // public onSearchNotify(event: KeyboardEvent): void {
```
Comment (kirillpanov) [inline src/app/courses/courses-list/courses-list.component.ts]: please remove commented code

File: src/app/courses/courses-list/courses-list.component.ts [hunk 6]
```diff
@@ -86,4 +92,20 @@ export class CoursesListComponent implements OnInit {
             refCount()
         );
     }
+
+    public ngAfterViewInit(): void {
+        const minQueryLength: number = 3;
+        const delayTime: number = 400;
+
+        this.searchComponent.notify
```
Comment (kirillpanov) [inline src/app/courses/courses-list/courses-list.component.ts]: do not forget to unsubscribe

File: src/app/shared/main-layout/main-layout.component.ts [hunk 7]
```diff
@@ -1,8 +1,27 @@
...
     styleUrls: ["./main-layout.component.scss"]
 })
-export class MainLayoutComponent {}
+export class MainLayoutComponent implements OnInit {
+    private loadingService: LoadingService;
+
+    /**
+     * Observable that is used to store Loader status
+     */
+    public loading: Observable<boolean>;
```
Comment (kirillpanov) [inline src/app/shared/main-layout/main-layout.component.ts]: is loading and loading$ if it is an observable

File: src/app/shared/main-layout/spinner/spinner.component.html [hunk 8]
```diff
@@ -0,0 +1,22 @@
+<div class="loading-screen-wrapper">
+    <div class="loading-screen-icon">
+        <div class="blobs">
+            <div class="blob-center"></div>
+            <div class="blob"></div>
```
Comment (kirillpanov) [inline src/app/shared/main-layout/spinner/spinner.component.html]: ngfor?

File: src/app/core/interceptors/loader-interceptor/loader.interceptor.ts (hunk 2 above)
Comment (gegorov) [reply src/app/core/interceptors/loader-interceptor/loader.interceptor.ts]: i forgot to remove it ) it was just for purpose of experimenting

File: src/app/core/services/loading-service/loading-service.service.ts (hunk 3 above)
Comment (gegorov) [reply src/app/core/services/loading-service/loading-service.service.ts]: fixed

File: src/app/core/services/loading-service/loading-service.service.ts (hunk 4 above)
Comment (gegorov) [reply src/app/core/services/loading-service/loading-service.service.ts]: fixed

File: src/app/courses/courses-list/courses-list.component.ts (hunk 5 above)
Comment (gegorov) [reply src/app/courses/courses-list/courses-list.component.ts]: done

File: src/app/courses/courses-list/courses-list.component.ts (hunk 6 above)
Comment (gegorov) [reply src/app/courses/courses-list/courses-list.component.ts]: fixed

File: src/app/shared/main-layout/main-layout.component.ts (hunk 7 above)
Comment (gegorov) [reply src/app/shared/main-layout/main-layout.component.ts]: fixed

**Pull request merged**